content/toolsAndResources/nge/ngeBlocks.md

```diff
@@ -0,0 +1,1384 @@
+---
+title: Node Geometry Editor Nodes
+image: 
+description: Available Nodes Within Node Geometry Editor
+keywords: geometry, procedural, non destructive, node, graph
+further-reading:
+video-overview:
+video-content:
+toc-levels: 2
+---
+
+## What to Expect
+This page is a breakdown of each node available in the Node Geometry Editor. The nodes will be listed in the same order by group as they are in the tool's node list. The reason for this order is that there may be some generalities that apply to all nodes in a group and those will be listed once rather than repeating them for each node. For each node we will discuss all inputs, outputs, properties, and considerations to ensure all requirements for using a node are highlighed. 
+
+## Sources
+Source nodes are a requirement for any Node Geometry graph as they supply the Geometry that is needed by the `GeometryOutput` node for the graph to be able to build. Most of these nodes will come with multiple inputs that will set the basic properties of the Geometry, but all have exactly one output which is Geometry generated by the source node. 
+
+Each parameter that can be set for the source Geometry can either be set by connecting a node or nodes to set a specific value or it can be set in the properties of the node. If a parameter's port on the node is connected by a wire, the parameter is removed from the parameter list on the node as it is deriving its value from the connected node(s). 
+
+If there is a set of connected properties - such as size, width, height, and depth - the more specific properties take precedence over the more general properties. In other words, if values are passed to size AND width then the parameter passed to width will apply to the width of the Geometry and height and depth will take their values from size. In this way, it is normal to see values wired to both general and specific properties at the same time. If, however, there are values passed to size, width, height, and depth then the value passed to size is not used at all and is not needed.
+
+![Box source node](/img/tools/nge/boxNode.jpg)

```

Comment:
You are correct and it bothered me too that the image wasn't seen from an anchor link. But I wanted to keep the image above the header tag as a visual break to help the reader realize we are at a new node. What I did instead was to call in Raanan and he is going to extend the way we link PGs in the docs to give me h tags with images that will act at anchor links. This will change the formatting of the markdown a bit, but it will solve the problem. He estimates that it will be in tomorrow so I will go in and update the doc once I have the new format.

Reply:
You are correct that many of the nodes now look different and the screen shots of all of them need to be updated. Unfortunately, this also applies to the annotated images from the general NGE doc as well. I had already updated them all once when icons and colors changed, but there are a lot of other visual changes that have also happened since. We are calling a break point now to update the initial docs and images (even if some are outdated) because trying to keep up with the pace of changes right now is a losing battle. We are going to come back in a few weeks once the updates and changes calm down a bit and update them all at once. 

As for the alignNode image, I am not sure why, but it appears to have been excluded from my initial commit. I will make sure it pushes with this one.